### DISCUSSION TITLE
How to remove trailing slash at the end of the url

### DISCUSSION THREAD

**Discussion by denistruhlaz** [nuxt/nuxt#9907] [Questions (Nuxt 2)]
Hello all,
is it possible to remove trailing slash at the end of the url without using 3rd part libs?
I'm struggling to remove "/" at the end of the every route in my nuxt project. 
- I checked documentation about nuxt.config and there is a property trailingSlash in router, I set that on False but without success, routes are still ending with the "/".
- I tried to do something with the extendRoutes also, but without success. 
- I tried to set the base path in router in nuxt config, no success.
- I tried configuring generate in nuxt config, setting subFolders to False
- I tired setting publicPath in generate instead of base path in router

Current behavior: 
http://localhost:3000/test-truli/
http://localhost:3000/test-truli/login/

Expected behavior:
 http://localhost:3000/test-truli
http://localhost:3000/test-truli/login

If someone have experience and willing to help I would appreciate that.

**Comment by nathanchase** [comment]
My solution to date has been to use "[@nuxtjs/redirect-module](https://github.com/nuxt-community/redirect-module)": "^0.3.1", and the following redirect rule:
```
// Remove trailing slash
  {
    from: '(?!^\/$|^\/[?].*$)(.*\/[?](.*)$|.*\/$)',
    to: (from, req) => {
      const base = req._parsedUrl.pathname.replace(/\/$/, '');
      const search = req._parsedUrl.search;
      return base + (search != null ? search : '');
    },
  },
```

**Reply by denistruhlaz** [reply]
Currently I'm trying to serve index.html from dist when user go on that specific route using AWS functionalities. If I don't succeed, I'll try this solution.

**Comment by kissu** [comment]
The first one is not working? Can you link us a repro?
Also, it's me or your `current` and `expected` behavior are the same?

**Reply by denistruhlaz** [reply]
yes, first solution that I tried was setting trailingSlash to false in router, but without success....
Current behavior have / at the end of every path, expected should NOT have / at the end of every path.

**Reply by kissu** [reply]
Do you have a repro for this?

**Reply by denistruhlaz** [reply]
yes I have, but unfortunately I cannot share it...

**Reply by kissu** [reply]
A repro is just you replicating the thing, it doesn't need to be your company's exact code. Without a repro, I'm not sure that we can help you anyhow.
